stores/facilitiesStore.ts

```diff
@@ -42,41 +49,81 @@ export const useFacilitiesStore = defineStore(
         }
 
         async function getFacilities() {
             const facilityResults = await queryFacilities()
             facilityData.value = facilityResults
         }
 
-        async function updateFacility(facility: MutationUpdateFacilityArgs) {
-            try {
-                return await graphQLClientRequestWithRetry(
-                    gqlClient.request.bind(gqlClient),
-                    updateExistingFacilityGqlMutation,
-                    facility
-                )
-            } catch (error) {
-                console.error('Failed to update facility:', error)
+        async function updateFacility():
+        Promise<ServerResponse<Maybe<Facility>>> {
+            const serverResponse = { data: {} as Maybe<Facility>, errors: [] as ServerError[], hasErrors: false }
+
+            const updateFacilityInput: MutationUpdateFacilityArgs = {
+                id: selectedFacilityId.value,
+                input: {
+                    contact: {
+                        address: {
+                            addressLine1En: facilitySectionFields.addressLine1En,
+                            addressLine1Ja: facilitySectionFields.addressLine1Ja,
+                            addressLine2En: facilitySectionFields.addressLine2En,
+                            addressLine2Ja: facilitySectionFields.addressLine2Ja,
+                            cityEn: facilitySectionFields.cityEn,
+                            cityJa: facilitySectionFields.cityJa,
+                            postalCode: facilitySectionFields.postalCode,
+                            prefectureEn: facilitySectionFields.prefectureEn,
+                            prefectureJa: facilitySectionFields.prefectureJa
+                        },
+                        email: facilitySectionFields.email,
+                        googleMapsUrl: facilitySectionFields.googlemapsURL,
+                        phone: facilitySectionFields.phone,
+                        website: facilitySectionFields.website
+                    },
+                    healthcareProfessionalIds: facilitySectionFields.healthProfessionalsRelations.length > 0
+                        ? facilitySectionFields.healthProfessionalsRelations
+                        : [],
+                    mapLatitude: parseFloat(facilitySectionFields.mapLatitude),
+                    mapLongitude: parseFloat(facilitySectionFields.mapLongitude),
+                    nameEn: facilitySectionFields.nameEn,
+                    nameJa: facilitySectionFields.nameJa
+                }
+            }
+
+            const response = await graphQLClientRequestWithRetry<Mutation>(
+                gqlClient.request.bind(gqlClient),
+                updateExistingFacilityGqlMutation,
+                updateFacilityInput
+            )
+
+            serverResponse.data = response.data?.updateFacility
+            serverResponse.errors = response.errors ? response.errors : []
```

Comment:
should add a log at least if there's errors

Reply:
A log is recorded here

```
export const graphQLClientRequestWithRetry = async <T>(
    gqlClientRequestFunction: (
        queryOrMutation: RequestDocument,
        variables?: unknown
    ) => Promise<T>,
    queryOrMutation: RequestDocument,
    variables: unknown,
    retryOptions?: graphQLClientRequestWithRetryOptions
): Promise<ServerResponse<T>> => {
    let attempts = 0
    const retryAmount = retryOptions?.retryAmount || 3
    const requestTimeoutInMilliseconds = retryOptions?.requestTimeoutInMilliseconds || 5000

    const executeGQLClientRequest = async (): Promise<ServerResponse<T>> => {
        try {
            const data = await gqlClientRequestFunction(queryOrMutation, variables)
            return { data, errors: [], hasErrors: false }
        } catch (error) {
            if (attempts < retryAmount) {
                attempts++
                if (attempts > 1) {
                    await new Promise(resolve => setTimeout(resolve, requestTimeoutInMilliseconds))
                }
                return executeGQLClientRequest() // retry request
            }

            // This is a consistent error messaging no matter the type of query or mutation
            console.error(`There was an error executing the request: ${error}`)
            const serverError = error as ServerErrorResponse

            // This map transforms errors if they exist
            const errors = serverError.response?.errors?.map(errorResponse => ({
                message: errorResponse.message,
                fieldWithError: errorResponse.locations,
                code: errorResponse.extensions.code as ErrorCode
            })) || []

            return { data: {} as T, errors, hasErrors: true }
        }
    }

    return executeGQLClientRequest()
}
```

It happens if there is every an error and there are user toast messages displayed for the specific error